src/identity/IdentityProviderHttpHandler.ts

```diff
@@ -1,52 +1,68 @@
+import { DataFactory } from 'n3';
+import urljoin from 'url-join';
 import type { ErrorHandler } from '../ldp/http/ErrorHandler';
+import type { RequestParser } from '../ldp/http/RequestParser';
+import { OkResponseDescription } from '../ldp/http/response/OkResponseDescription';
+import { RedirectResponseDescription } from '../ldp/http/response/RedirectResponseDescription';
+import type { ResponseDescription } from '../ldp/http/response/ResponseDescription';
 import type { ResponseWriter } from '../ldp/http/ResponseWriter';
+import type { Operation } from '../ldp/operations/Operation';
+import { BasicRepresentation } from '../ldp/representation/BasicRepresentation';
+import type { RepresentationPreferences } from '../ldp/representation/RepresentationPreferences';
 import { getLoggerFor } from '../logging/LogUtil';
 import type { HttpHandlerInput } from '../server/HttpHandler';
 import { HttpHandler } from '../server/HttpHandler';
 import type { HttpRequest } from '../server/HttpRequest';
 import type { HttpResponse } from '../server/HttpResponse';
-import type { TemplateHandler } from '../server/util/TemplateHandler';
+import type { RepresentationConverter } from '../storage/conversion/RepresentationConverter';
+import { APPLICATION_JSON } from '../util/ContentTypes';
 import { BadRequestHttpError } from '../util/errors/BadRequestHttpError';
 import { assertError, createErrorMessage } from '../util/errors/ErrorUtil';
 import { InternalServerError } from '../util/errors/InternalServerError';
 import { trimTrailingSlashes } from '../util/PathUtil';
+import { CONTENT_TYPE_TERM, SOLID_META } from '../util/Vocabularies';
 import type { ProviderFactory } from './configuration/ProviderFactory';
 import type {
   Interaction,
   InteractionHandler,
-  InteractionHandlerResult,
+  InteractionHandlerResult, InteractionResponseResult,
 } from './interaction/email-password/handler/InteractionHandler';
 import { IdpInteractionError } from './interaction/util/IdpInteractionError';
 import type { InteractionCompleter } from './interaction/util/InteractionCompleter';
+import namedNode = DataFactory.namedNode;
```

Comment:
Without having studied the details: that is a ton of diverse imports.

Is that a sign that we might need to split off some functionality into another class?

Reply:
The RDF stuff is what seems too much (`DataFactory`). Perhaps a metadata helper somewhere?

Reply:
> Is that a sign that we might need to split off some functionality into another class?

Probably, I already give some ideas above of how this could be done, but my plan was to first get the main behaviour right and then split it off. Not exactly the best architectural design method, but it doesn't impact any other components for now and I felt there already were plenty of changes here for 1 PR.

> The RDF stuff is what seems too much (DataFactory). Perhaps a metadata helper somewhere?

This is a common problem in the entire project actually, many places where we need RDF terms have this. Also related: we actually have 2 libraries for creating terms: `N3.js` and `@rdfjs/data-model`. Having a metadata helper somewhere to have a consistent solution for this would probably be good (and would allow us to remove the second dependency). Would create a separate issue for this instead of fixing this here though.